cmake/onnxruntime.cmake

```diff
@@ -40,10 +40,7 @@ else()
 endif()
 
 if (NOT WIN32)
-  set(CMAKE_SKIP_BUILD_RPATH  FALSE)
-  set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
-  set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
-  set(CMAKE_INSTALL_RPATH "$ORIGIN/")
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-rpath='$ORIGIN'")
```

Comment:
If this is needed by the C# binding, can we only apply it to c#?
One big difference between Windows and Linux is:  Linux won't automatically load a binary from current dir. current dir is not in the PATH env, or LD_LIBRARY_PATH env.   Even it can be changed, nobody want to make Linux working like Windows.  If we distribute our binary with such a hidden setting("-rpath='$ORIGIN'"),  I guess most of the Linux C/C++ users won't like it.